@here/harp-datasource-protocol/lib/Expr.ts

```diff
@@ -521,6 +575,14 @@ export class MatchExpr extends Expr {
     accept<Result, Context>(visitor: ExprVisitor<Result, Context>, context: Context): Result {
         return visitor.visitMatchExpr(this, context);
     }
+
+    protected exprIsDynamic() {
+        return (
+            this.value.isDynamic() ||
+            this.branches.some(([_, branch]) => branch.isDynamic()) ||
```

Comment:
if the value is not dynamic the expression would be only dynamic if the according match would be dynamic

Reply:
good point, I will restrict the behavior of `isDynamic` for `Case` and `Match` in a follow up PR.